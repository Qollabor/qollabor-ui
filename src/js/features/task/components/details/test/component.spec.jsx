/* eslint-disable react/prop-types */
/* eslint-disable react/no-multi-comp */

import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';


describe('features/task/taskDetails', () => {
  beforeEach(() => {
    class ReactComponent extends React.Component {
      render () {
        return (<div />);
      }
    }

    jest.mock('../../../../../components/capsules', () => ({ StatusCapsule: ReactComponent }));
    jest.mock('material-ui/RaisedButton', () => ReactComponent);
    jest.mock('material-ui/FlatButton', () => ReactComponent);
    jest.mock('../../breadcrumb', () => ({ TaskBreadcrumb: ReactComponent }));
    jest.mock('../../schema-form', () => ({ formData }) => (<div formData={formData} />));
    jest.mock('../../../components/action-buttons', () => ({ ActionButtons: ReactComponent }));
  });

  it('has correct default options', () => {
    const TaskDetails = require('../taskDetails').default; // eslint-disable-line global-require

    const caseId = 'blah';
    const taskId = '123';
    const isFetching = false;
    const taskDetails = { assignee: 'wilbur', owner: 'admin' };
    const component = renderer.create(
      <MuiThemeProvider>
        <TaskDetails
          caseId={caseId}
          taskId={taskId}
          isFetching={isFetching}
          taskDetails={taskDetails}
        />
      </MuiThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('implements raw output', () => {
    const TaskDetails = require('../taskDetails').default; // eslint-disable-line global-require

    const caseId = 'blah';
    const taskId = '123';
    const isFetching = false;
    const taskDetails = { assignee: 'wilbur', owner: 'admin', rawOutput: { raw: 'output' } };
    const component = renderer.create(
      <MuiThemeProvider>
        <TaskDetails
          caseId={caseId}
          taskId={taskId}
          isFetching={isFetching}
          taskDetails={taskDetails}
        />
      </MuiThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('handles initTenantOrganisations prop', () => {
  //   const TenantOrganisations = require('../component').default; // eslint-disable-line global-require
  //
  //   const data = [];
  //   const initTenantOrganisations = () => {};
  //   const component = renderer.create(
  //     <MuiThemeProvider>
  //       <TenantOrganisations
  //         data={data}
  //         initTenantOrganisations={initTenantOrganisations}
  //       />
  //     </MuiThemeProvider>
  //   );
  //   const tenantOrganisationComponent = component.toJSON();
  //   expect(tenantOrganisationComponent).toMatchSnapshot();
  // });
});
