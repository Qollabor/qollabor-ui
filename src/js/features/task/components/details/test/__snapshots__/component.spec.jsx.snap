exports[`features/task/taskDetails has correct default options 1`] = `
<div>
  <div
    style={
      Object {
        "float": "right",
      }
    }>
    <div />
  </div>
  <div
    style={
      Object {
        "float": "right",
      }
    }>
    <div />
  </div>
  <div />
  <div>
    <section
      style={
        Object {
          "marginBottom": "10px",
          "paddingLeft": "10px",
        }
      }>
      <div
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 15,
            "width": "100%",
          }
        }>
        <div
          style={
            Object {
              "height": 60,
            }
          }>
          <div
            style={
              Object {
                "float": "right",
              }
            }>
            <br />
          </div>
          <div
            style={
              Object {
                "fontSize": 14,
                "height": 10,
              }
            }>
            <span
              style={
                Object {
                  "lineHeight": "35px",
                  "marginTop": 5,
                }
              }>
              <span
                style={
                  Object {
                    "fontSize": 14,
                  }
                }>
                Assigned to
              </span>
              <span
                style={
                  Object {
                    "fontSize": 14,
                    "fontWeight": 500,
                    "marginLeft": 5,
                  }
                }
                title={undefined}>
                wilbur
              </span>
            </span>
            <br />
            <span
              style={
                Object {
                  "lineHeight": "35px",
                  "marginTop": 5,
                }
              }>
              <span
                style={
                  Object {
                    "fontSize": 14,
                  }
                }>
                Owned by
              </span>
              <span
                style={
                  Object {
                    "fontSize": 14,
                    "fontWeight": 500,
                    "marginLeft": 5,
                  }
                }
                title={undefined}>
                admin
              </span>
            </span>
            <div
              style={
                Object {
                  "display": "none",
                }
              } />
          </div>
        </div>
      </div>
    </section>
  </div>
  <div
    formData={undefined} />
</div>
`;

exports[`features/task/taskDetails implements raw output 1`] = `
<div>
  <div
    style={
      Object {
        "float": "right",
      }
    }>
    <div />
  </div>
  <div
    style={
      Object {
        "float": "right",
      }
    }>
    <div />
  </div>
  <div />
  <div>
    <section
      style={
        Object {
          "marginBottom": "10px",
          "paddingLeft": "10px",
        }
      }>
      <div
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 15,
            "width": "100%",
          }
        }>
        <div
          style={
            Object {
              "height": 60,
            }
          }>
          <div
            style={
              Object {
                "float": "right",
              }
            }>
            <br />
          </div>
          <div
            style={
              Object {
                "fontSize": 14,
                "height": 10,
              }
            }>
            <span
              style={
                Object {
                  "lineHeight": "35px",
                  "marginTop": 5,
                }
              }>
              <span
                style={
                  Object {
                    "fontSize": 14,
                  }
                }>
                Assigned to
              </span>
              <span
                style={
                  Object {
                    "fontSize": 14,
                    "fontWeight": 500,
                    "marginLeft": 5,
                  }
                }
                title={undefined}>
                wilbur
              </span>
            </span>
            <br />
            <span
              style={
                Object {
                  "lineHeight": "35px",
                  "marginTop": 5,
                }
              }>
              <span
                style={
                  Object {
                    "fontSize": 14,
                  }
                }>
                Owned by
              </span>
              <span
                style={
                  Object {
                    "fontSize": 14,
                    "fontWeight": 500,
                    "marginLeft": 5,
                  }
                }
                title={undefined}>
                admin
              </span>
            </span>
            <div
              style={
                Object {
                  "display": "none",
                }
              } />
          </div>
        </div>
      </div>
    </section>
  </div>
  <div
    formData={
      Object {
        "raw": "output",
      }
    } />
</div>
`;
